sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -83,13 +78,21 @@ export class DefaultManagedIdentityCredential extends ManagedIdentityCredential
       ?.managedIdentityResourceId;
 
     // ManagedIdentityCredential throws if both the resourceId and the clientId are provided.
-    const managedIdentityOptions: ManagedIdentityCredentialOptions = {
-      resourceId: managedResourceId,
-      clientId: managedIdentityClientId,
-      ...options,
-    };
-
-    super(managedIdentityOptions);
+    // let managedIdentityOptions;
+    if (managedResourceId) {
+      const managedIdentityResourceIdOptions: ManagedIdentityCredentialResourceIdOptions = {
+        ...options,
+      };
+      managedIdentityResourceIdOptions.resourceId = managedResourceId;
+      super(managedIdentityResourceIdOptions);
+    }
+    if (managedIdentityClientId) {
```

Comment:
Shouldn't this be an `else if`? You may run into trouble calling `super` twice if someone passes both client and resource ID at runtime.

Reply:
updated